File: [242]有效的字母异位词.cpp
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> count(26, 0);
        for (char c : s)
            count[c-'a'] += 1;
        for (char c : t)
            count[c-'a'] -= 1;

        for (int n : count)
            if (n != 0)       
                return false;
        return true;
    }
};

File: solutions/[242]有效的字母异位词.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        c2count = dict()
        for c in s:
            count = c2count.get(c, 0)
            c2count[c] = count + 1
        
        for c in t:
            count = c2count.get(c, 0)
            if count == 0:
                return False
            else:
                c2count[c] = count - 1
        
        return True